homeassistant/components/netatmo/sensor.py

```diff
@@ -73,23 +74,54 @@
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the available Netatmo weather sensors."""
     netatmo = hass.components.netatmo
-    data = NetAtmoData(netatmo.NETATMO_AUTH, config.get(CONF_STATION, None))
 
     dev = []
+    if CONF_MODULES in config:
+        manual_config(netatmo, config, dev)
+    else:
+        auto_config(netatmo, config, dev)
+
+    if dev:
+        add_entities(dev, True)
+
+
+def manual_config(netatmo, config, dev):
+    """Handle manual configuration."""
     import pyatmo
-    try:
-        if CONF_MODULES in config:
+
+    not_handled = []
+    for data_class in [pyatmo.WeatherStationData]:
+        data = NetAtmoData(netatmo.NETATMO_AUTH, data_class,
+                           config.get(CONF_STATION, None))
+        try:
             # Iterate each module
             for module_name, monitored_conditions in \
                     config[CONF_MODULES].items():
                 # Test if module exists
                 if module_name not in data.get_module_names():
-                    _LOGGER.error('Module name: "%s" not found', module_name)
+                    if module_name not in not_handled:
+                        not_handled.append(module_name)
                     continue
+                elif module_name in not_handled:
+                    not_handled.remove(module_name)
                 # Only create sensors for monitored properties
                 for variable in monitored_conditions:
                     dev.append(NetAtmoSensor(data, module_name, variable))
-        else:
+        except pyatmo.NoDevice:
+            continue
+
+    for module_name in not_handled:
+        _LOGGER.error('Module name: "%s" not found', module_name)
+
+
+def auto_config(netatmo, config, dev):
+    """Handle auto configuration."""
+    import pyatmo
+
+    for data_class in [pyatmo.WeatherStationData, pyatmo.HomeCoachData]:
+        data = NetAtmoData(netatmo.NETATMO_AUTH, data_class,
+                           config.get(CONF_STATION, None))
```

Comment:
Remove `None`, it is default for `get()`

Reply:
Fixed